docs/proposals/sidecar.md

```diff
@@ -0,0 +1,41 @@
+# Executes Commands in a Local Pod from Kubelet
+
+Currently Kubelet uses [exec interface](https://github.com/kubernetes/kubernetes/tree/master/pkg/util/exec) to invoke executables that are installed on the host. Packages that contain the executables must be installed on the nodes and produce the same output that `exec` callers expect. As a result, package installation, version tracking, and upgrade complicate both Kubernetes configuration and development.
+
+This proposal aims to containerize command in a local Pod and thus eliminates the host-package dependency. Kubelet can pull container images that have the required executables and run the executables inside the local Pod. 
+
+## Use Cases
+
+### Volume Plugins
+* rbd volume plugin currently executes `rbd` command that is available in `ceph-common` package. Containerized `rbd` can execute `rbd` command from [ceph/base](https://github.com/ceph/ceph-docker/tree/master/base) container image.
+* iSCSI volume plugin executes `iscsiadmin` command to configure iSCSI devices.
+* Glusterfs volume plugin requires glusterfs-fuse package that provides mount helper to mount a Glusterfs volume. 
+
+
+## Utilities that Vary on Different OSes
+Some packages that Kubernetes use are missing on some OSes. [A recent example](https://github.com/kubernetes/kubernetes/pull/14109) illustrates that utilities used by some OSes are missing on other platforms. In such case, a dynamically pulled containers can resolve such differences.
+
+## Scope
+
+Local Pods are created and scheduled by Kubelet, rather than API server. Local Pods can be long running or exit after command is finished. Long running Pods are separately discussed in [Daemon Pod](https://github.com/kubernetes/kubernetes/issues/1518). This proposal focuses on short-lived Pods that exit after command is finished.
+
+## Structure
+A proposed interface to invoke a local Pod and execute a command is as the following:
```

Comment:
I don't understand the proposal. This does exec into a container or launches a container? Does it create the pod as well?

We're going to need the ability to launch containers in the context of a pod, such as to initialize volumes #831, and in the context of the host, such as to reconfigure the host on demand on behalf of an application ("config packages" for Borgonauts). This sounds like the latter case. If so, why not just spawn a pod?